messagix/client.go

```diff
@@ -220,6 +225,7 @@ func (c *Client) Connect() error {
 		connectionAttempts := 1
 		reconnectIn := 2 * time.Second
 		for {
+			c.disableSendingMessages() // In case we're reconnecting from a normal network error
```

Comment:
This should probably be after Connect returns, otherwise it stays enabled until the sleep at the end of the loop is done

Reply:
🤔 Did you mean "stays disabled"? 

My idea for putting it here was that as soon as we go into this `for` loop, we know we need to connect / reconnect, and therefore there is no reason to have sending messages (`canSendMessages`) enabled (`true`). Therefore we better set it to false asap until the connection is completed, and the `EnableSendingMessages` have been called from `handleReadyEvent`.

Is there any other angle I'm not seeing? 

Thanks!

Reply:
Gotcha, that makes sense. Thank you for the clarification. Just moved the `disableSendingMessages`.

Off topic, imho the code reads a bit weird

```golang
// client.go
func (c *Client) Connect() error {
	// ...
	go func() {
		// ...
		for {
			connectStart := time.Now()
			err := c.socket.Connect()
			c.disableSendingMessages()
			// ...
		}
	}()
	// ...
}
```

When I read this code for the first tiem, I had no idea that `c.socket.Connect()` would block in the socket's `readLoop`. Someone new to the code could be like, _why are we disabling sending messages right after connecting?_. From the outside, it's not obvious that the execution will only move forward once there is an error in the connection. I guess this is something we can improve in the future.